src/psyclone/psyir/backend/sympy_writer.py

```diff
@@ -788,7 +789,20 @@ def intrinsiccall_node(self, node: IntrinsicCall) -> str:
             args = self._gen_arguments(node)
             return f"{self._nindent}{name}({args})"
         except KeyError:
-            return super().intrinsiccall_node(node)
+            # This section is copied from FortranWriter IntrinsicCall,
+            # but doesn't attempt to match argument names and so avoids
+            # readding optional argument names back in.
+            args = self._gen_arguments(node)
+            if node.routine.name not in [
+                    "DATE_AND_TIME", "SYSTEM_CLOCK", "MVBITS",
+                    "RANDOM_NUMBER", "RANDOM_SEED"]:
+                # Most intrinsics are functions and so don't have 'call'.
+                if not node.parent or isinstance(node.parent, Schedule):
+                    return f"{self._nindent}{node.routine.name}({args})\n"
+                return f"{node.routine.name}({args})"
+            if not node.parent or isinstance(node.parent, Schedule):
+                return (f"{self._nindent}call "
+                        f"{self._visit(node.routine)}({args})\n")
 
```

Comment:
As with the code in FortranWriter, it looks like this could be simplified. It also looks as though it's possible to reach the end of this method and so return None?